solr/core/src/java/org/apache/solr/storage/SizeAwareDirectory.java

```diff
@@ -121,50 +181,56 @@ public long size() throws IOException {
       final String[] files = in.listAll();
 
       LongAdder recomputeSize = new LongAdder();
+      LongAdder recomputeOnDiskSize = new LongAdder();
       Set<String> recomputed = ConcurrentHashMap.newKeySet();
-      LongObjectProcedure<String> dualSizeWriter =
-          (fileSize, name) -> {
+      SizeWriter dualSizeWriter =
+          (fileSize, onDiskFileSize, name) -> {
             size.add(fileSize);
+            onDiskSize.add(onDiskFileSize);
             if (fileSize >= 0 || recomputed.remove(name)) {
               // if it's a removal, we only want to adjust if we've already
               // incorporated this file in our count!
               recomputeSize.add(fileSize);
+              recomputeOnDiskSize.add(onDiskFileSize);
             }
           };
       sizeWriter = dualSizeWriter;
       for (final String file : files) {
-        long fileSize;
+        Sizes sizes;
         recomputed.add(file);
         SizeAccountingIndexOutput liveOutput = liveOutputs.get(file);
         if (liveOutput != null) {
           // get fileSize already written at this moment
-          fileSize = liveOutput.setSizeWriter(dualSizeWriter);
+          sizes = liveOutput.setSizeWriter(dualSizeWriter);
         } else {
-          fileSize = DirectoryFactory.sizeOf(in, file);
+          long fileSize = DirectoryFactory.sizeOf(in, file);
+          long onDiskFileSize = DirectoryFactory.onDiskSizeOf(in, file);
+          sizes = new Sizes(fileSize, onDiskFileSize);
           if (fileSize > 0) {
             // whether the file exists or not, we don't care about it if it has zero size.
             // more often though, 0 size means the file isn't there.
-            fileSizeMap.put(file, fileSize);
+            fileSizeMap.put(file, new Sizes(fileSize, onDiskFileSize));
```

Comment:
nit: I think we can just use `sizes` (defined above) here, rather than creating a new `Sizes` instance?